pyxrf/model/fit_spectrum.py

```diff
@@ -256,17 +274,222 @@ def energy_bound_low_update(self, change):
     @observe('selected_index')
     def _selected_element_changed(self, change):
         if change['value'] > 0:
-            selected_element = self.element_list[change['value']-1]
+            ind_sel = change['value'] - 1
+            if ind_sel >= len(self.element_list):
+                ind_sel = len(self.element_list) - 1
+                self.selected_index = ind_sel + 1  # Change the selection as well
+            selected_element = self.element_list[ind_sel]
             if len(selected_element) <= 4:
                 element = selected_element.split('_')[0]
                 self.elementinfo_list = sorted([e for e in list(self.param_dict.keys())
                                                 if (element+'_' in e) and  # error between S_k or Si_k
                                                 ('pileup' not in e)])  # Si_ka1 not Si_K
-                print(self.elementinfo_list)
+                logger.info(f"Element line info: {self.elementinfo_list}")
             else:
                 element = selected_element  # for pileup peaks
                 self.elementinfo_list = sorted([e for e in list(self.param_dict.keys())
                                                 if element.replace('-', '_') in e])
+                logger.info(f"User defined or pileup peak info: {self.elementinfo_list}")
+
+    def _compute_fwhm_base(self):
+        # Computes 'sigma' value based on default parameters and peak energy (for Userpeaks)
+        #   does not include corrections for fwhm
+        # If both peak center (energy) and fwhm is updated, energy needs to be set first,
+        #   since it is used in computation of ``fwhm_base``
+
+        v_const = 2 * np.sqrt(2 * np.log(2))
```

Comment:
Can `v_const` be calculated once somewhere to avoid  duplication of the same code? I saw at least 2 more places below doing the same operation.

Reply:
Fixed